Fix activate function return type in TS defs

perryh opened the pull request:
addresses https://github.com/DataDog/dd-trace-js/issues/480

rochdev commented on docs/test.ts:
```diff
@@ -143,7 +143,7 @@ const scope = tracer.scope()
 span = scope.active();
 
-scope.activate(span, () => {});
+const activateType: string | void = scope.activate(span, () => 'test');
```
Probably better to test `string` and `void` individually.

perryh commented:
I think I found some other types that need to be updated. hold on.

perryh commented:
@rochdev ready for review. I fixed `activate` to return only one type instead of "T | void".

Also fixed `bind` so that the function it returns only returns one type instead of "T | void".

rochdev commented on docs/index.d.ts:
```diff
@@ -270,7 +270,8 @@ export declare interface Scope {
    * @returns The bound target.
    */
-  bind<T>(fn: ((...args: any[]) => T | void), span?: Span | null): ((...args: any[]) => T | void);
+  bind<T extends (...args: any[]) => void>(fn: T, span?: Span | null): T;
```
Why is this one necessary in addition to the one below?

perryh replied:
This was confusing to me as well. If I remove this line, the TS compiler complains and seems to match the function that returns void passed in the `bindFunctionVoidType` example to the bind definition accepting an EventEmitter.

rochdev replied:
It seems to work with the old syntax without `void` as well, but I guess your version preserves the original type of the function. I think `tracer.wrap()` may has the same limitation. If you have some time, it would be awesome to fix it in the same PR, otherwise no worries I'll make another one later :)

perryh replied:
@rochdev I will defer to you :) have not yet upgraded to 0.10.
